### PR TITLE
⚡ Upload sqlite file in separate process

### PR DESCRIPTION
**Pull request by falexwolf**
- https://laminlabs.slack.com/archives/C04FPE8V01W/p1686136203501999

### CODE REVIEW

File: lamindb_setup/dev/_settings_instance.py [hunk 1]
```diff
@@ -98,17 +106,21 @@ def _sqlite_file_local(self) -> Path:
...
+            cache_file = self.storage.cloud_to_local_no_update(cloud_file)
+
+            if new_process:
+                from multiprocessing import Process
+
+                Process(
+                    target=upload_and_sync_timestamp, args=(cache_file, cloud_file)
+                ).start()
+            else:
+                upload_and_sync_timestamp(cache_file, cloud_file)
```
Comment (falexwolf) [inline lamindb_setup/dev/_settings_instance.py:123]: @Koncopd!

**Comment by Koncopd** [comment]
There is a problem with the sync notebook, but i don't really understand why this happens now, the update of local sqlite file should not be affected.

**Comment by Koncopd** [comment]
Now it is probably better not to run this notebook at all.

**Comment by Koncopd** [comment]
image
Running locally it fails on Windows. I have no idea why it still work on linux / CI on main.

**Comment by falexwolf** [comment]
I reproduce the error on CI locally on my mac depending on how fast I run the notebook 😆 - if I run it slowly and allow the upload process to finish everything is good. If I run it fast and the upload process hasn't yet finished and the time stamp is written, it evidently fails
image

This makes me think: OMG, this is dangerous without any await logic among processes.

And that makes me think: OMG, this is going to be complicated...

**Pull request closed without merging**